src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/RSA/SignVerify.cs

```diff
@@ -497,22 +498,26 @@ public void VerifySignature_SHA256_2048()
             VerifySignature(signature, TestData.HelloBytes, "SHA256", TestData.RSA2048Params);
         }
 
-        [Fact]
-        public void SignAndVerify_SHA1_1024()
-        {
-            SignAndVerify(TestData.HelloBytes, "SHA1", TestData.RSA1024Params);
-        }
-
-        [Fact]
-        public void SignAndVerify_SHA1_2048()
+        [Theory]
+        [MemberData(nameof(RoundTripTheories))]
+        public void SignAndVerify_Roundtrip(string hashAlgorithm, RSAParameters rsaParameters)
         {
-            SignAndVerify(TestData.HelloBytes, "SHA1", TestData.RSA2048Params);
+            SignAndVerify(TestData.HelloBytes, hashAlgorithm, rsaParameters);
         }
 
-        [Fact]
-        public void SignAndVerify_SHA256_1024()
+        public static IEnumerable<object[]> RoundTripTheories
         {
-            SignAndVerify(TestData.HelloBytes, "SHA256", TestData.RSA1024Params);
+            get
+            {
+                yield return new object[] { nameof(HashAlgorithmName.MD5), TestData.RSA1024Params };
+                yield return new object[] { nameof(HashAlgorithmName.MD5), TestData.RSA2048Params };
+                yield return new object[] { nameof(HashAlgorithmName.SHA1), TestData.RSA1024Params };
+                yield return new object[] { nameof(HashAlgorithmName.SHA1), TestData.RSA2048Params };
+                yield return new object[] { nameof(HashAlgorithmName.SHA256), TestData.RSA1024Params };
+                yield return new object[] { nameof(HashAlgorithmName.SHA256), TestData.RSA2048Params };
+                yield return new object[] { nameof(HashAlgorithmName.SHA384), TestData.RSA2048Params };
+                yield return new object[] { nameof(HashAlgorithmName.SHA512), TestData.RSA2048Params };
+            }
         }
```

Comment:
Maybe a tad more clear as:
```C#
foreach (RsaParameters rsaParameters in new[] { TestData.RSA1024Params, TestData.RSA2048Params })
{
    yield return new object[] { nameof(HashAlgorithmName.MD5), rsaParameters };
    yield return new object[] { nameof(HashAlgorithmName.SHA1), rsaParameters };
    yield return new object[] { nameof(HashAlgorithmName.SHA256), rsaParameters };
}
yield return new object[] { nameof(HashAlgorithmName.SHA384), TestData.RSA2048Params };
yield return new object[] { nameof(HashAlgorithmName.SHA512), TestData.RSA2048Params };
```
?

Reply:
I don't feel strongly one way or the other, if you are leaning toward changing it, I'd be happy to do so.

Reply:
I have a small preference for my suggestion, but it's not a big deal.